### ISSUE TITLE
Misaligned Glyphs - Patch Myself? 

### ISSUE THREAD

**Issue by her** [ryanoasis/vim-devicons#126]
[![https://gyazo.com/fa5b1e4bea796a1c428007401ef2604c](https://i.gyazo.com/fa5b1e4bea796a1c428007401ef2604c.png)](https://gyazo.com/fa5b1e4bea796a1c428007401ef2604c)

Any idea what's causing this? 

I tried setting Non-ASCII font to 12 (my default font is 13) that only half fixed the issue but broke the other glyphs 

[![https://gyazo.com/b64128b939121388a63bd3dc7c9029ca](https://i.gyazo.com/b64128b939121388a63bd3dc7c9029ca.png)](https://gyazo.com/b64128b939121388a63bd3dc7c9029ca)

I'm currently using Nerd-Font Sauce Code Pro Regular size 13... I'm getting this error on Mono as well. 

If I use Sauce Code Pro Powerline I don't have this issue, but of course this breaks my amazing devicons 

[![https://gyazo.com/dc3fb7ea4f4940682a0b86104dab65d2](https://i.gyazo.com/dc3fb7ea4f4940682a0b86104dab65d2.png)](https://gyazo.com/dc3fb7ea4f4940682a0b86104dab65d2)

### tl;dr

How can I take the glyphs from Source Code Pro Powerline and apply them to the nerd font?

**Comment by ryanoasis** [comment]
The patcher is not supposed to override Powerline   glyphs if the font contains  it already but seems to not be working in this case.  https://github.com/ryanoasis/nerd-fonts/blob/master/gotta-patch-em-all-font-patcher!.sh#L65

I'll try looking into this, maybe tomorrow. Thanks

**Comment by her** [comment]
Well I should mention I didn't patch Source Code Pro-Powerline. 

I downloaded your patched fonts, moved the ones I needed to `~/Library/Fonts/` 

[![https://gyazo.com/fa40d059c535f0702aa96113f4835ac2](https://i.gyazo.com/fa40d059c535f0702aa96113f4835ac2.png)](https://gyazo.com/fa40d059c535f0702aa96113f4835ac2)

then within iTerm selected your font

[![https://gyazo.com/a315518d981dac2923aab6afcc9ca865](https://i.gyazo.com/a315518d981dac2923aab6afcc9ca865.png)](https://gyazo.com/a315518d981dac2923aab6afcc9ca865)

and added the appropriate settings to my .vimrc

**Comment by her** [comment]
@ryanoasis 

here's a comparison of E0B0 from Source Code Pro-Powerline and Nerd Fonts 

(both are regular) 


[![https://gyazo.com/70cfefa8fd6f19973a9a066c2cf67dfc](https://i.gyazo.com/70cfefa8fd6f19973a9a066c2cf67dfc.png)](https://gyazo.com/70cfefa8fd6f19973a9a066c2cf67dfc)

There does seem to be a bit of difference here

**Comment by her** [comment]
Another update... 

[![https://gyazo.com/34eedb3107964b7b73f058e73a34a724](https://i.gyazo.com/34eedb3107964b7b73f058e73a34a724.png)](https://gyazo.com/34eedb3107964b7b73f058e73a34a724)

Changing **Regular Font** within iTerm to 14pt and keeping Non-ASCII at 13pt fixes the misaligned glyphs. 

[![https://gyazo.com/ad4f9448d5370d7fc921ee30eb8f0940](https://i.gyazo.com/ad4f9448d5370d7fc921ee30eb8f0940.png)](https://gyazo.com/ad4f9448d5370d7fc921ee30eb8f0940)

This is sort of a hack though since it's changing my global font size entirely :( 

I also tried patching your font with the glyphs from Nerd-Font Sauce Code Pro but it failed so terribly I didn't even bother to take a screen shot. lol

Maybe the problem was that Sauce Code Pro-Powerline is .otf and and Nerd-Font Sauce Code Pro .ttf?

**Comment by ryanoasis** [comment]
Thanks for the great detail truly :+1: 

It will be helpful to me to look into issues with the fonts.

> I downloaded your patched fonts, moved the ones I needed to ~/Library/Fonts/

Ah I mean these fonts also just use the same patcher I mentioned above. If the unpatched version is okay then my guess is it is not ignoring the powerline glyphs as it should be.

**Comment by beanaroo** [comment]
I believe this could be a problem with the powerline glyphs provided by Adobe Source Code Pro, [referenced here](https://github.com/adobe-fonts/source-code-pro/issues/43#issuecomment-93478072).

It's noted that font anti-aliasing/subpixel rendering/hinting settings and terminal used may complicate things.

Personally, I have Sauce Code Pro working on Termite (VTE) and it can fall back to the other patched font for devicon glyphs. As ryan is aware, I have also reported it to the nerd-fonts package [here](https://github.com/ryanoasis/nerd-fonts/issues/60#issuecomment-190788034).

**Comment by ryanoasis** [comment]
Okay so I am going to close this since now [Nerd Fonts](https://github.com/ryanoasis/nerd-fonts) will **not** overwrite any powerline glyphs if they are present in the original source font ([see the commit here](https://github.com/ryanoasis/nerd-fonts/commit/1607658bc39e66ffa19d9e0e12b0a31f2b963b2f)).

See below:

![selection_15_10_16_21 46 17_1917x1073_001](https://cloud.githubusercontent.com/assets/8083459/19414644/f95cf18a-9320-11e6-814e-9419aad6f1b8.png)

Feel free to re-open if this is a larger question but either way I think this belongs on [Nerd Fonts](https://github.com/ryanoasis/nerd-fonts) :smile:
